code/modules/power/turbine/turbine_computer.dm

```diff
@@ -0,0 +1,100 @@
+/obj/machinery/computer/turbine_computer
+	name = "gas turbine control computer"
+	desc = "A computer to remotely control a gas turbine."
+	icon_screen = "turbinecomp"
+	icon_keyboard = "tech_key"
+	circuit = /obj/item/circuitboard/computer/turbine_computer
+	///Reference of the connected machine to this computer
+	var/obj/machinery/power/turbine/core_rotor/main_control
```

Comment:
I think you should use a weakref for this, since the var's only used in a few places and is easy to manage

Reply:
no idea how to use a weakref (as said above) i'll try to learn but make no promises

Reply:
you figure it out?

Reply:
mmaybe